packages/cli/src/index.ts

```diff
@@ -20,30 +21,28 @@ program.parse();
 
 const options = program.opts();
 
-if (program.args[0] === 'upload') {
-  uploadS3Snapshot(path.resolve(process.cwd(), options.snapshot));
-} else if (program.args[0] === 'build') {
-  if (options.config !== undefined) {
-    const config = await import(
-      pathToFileURL(path.resolve(process.cwd(), options.config)).toString()
-    );
-    if (!config) {
-      throw new Error(
-        `Could not find config file at ${path.resolve(process.cwd(), options.config)}.`
-      );
-    }
-    build(config.default, path.resolve(process.cwd(), options.out), options);
-  }
-} else if (program.args[0] === 'serve') {
-  if (options.config !== undefined) {
-    const config = await import(
-      pathToFileURL(path.resolve(process.cwd(), options.config)).toString()
+let config;
+if (options.config !== undefined) {
+  config = await import(pathToFileURL(path.resolve(process.cwd(), options.config)).toString());
+
+  if (!config) {
+    throw new Error(
+      `[Mosaic] could not find config file at ${path.resolve(process.cwd(), options.config)}.`
     );
-    if (!config) {
-      throw new Error(
-        `Could not find config file at ${path.resolve(process.cwd(), options.config)}.`
-      );
-    }
-    serve(config.default, options.port, options.scope && options.scope.split(','));
   }
+} else {
+  throw new Error(`[Mosaic] no config file provided`);
+}
+
+if (program.args[0] === 'build') {
+  build(config.default, path.resolve(process.cwd(), options.out), options);
+}
+if (program.args[0] === 'serve') {
+  serve(config.default, options.port, options.scope && options.scope.split(','));
+}
+if (program.args[0] === 'upload') {
```

Comment:
@DavieReid upload and deploy are the same thing, albeit on different platforms ?

Reply:
Small tidy up required?
updateTraceFile feels vercel specific..
the current `upload-snapshot-s3` is equally s3 specific.
Do we just rename these files and call a 'deploy' method or similar?

Reply:
Yeah we should revisit this...